### PR TITLE
Bridge Mvt -> Eth transfer e2e Test

### PR DESCRIPTION
**Pull request by musitdev**
# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.

Mvt -> Eth Transfer e2e test.
# Changelog
Update the contract deployment and add the initialize call to the counterpart contract.
Add the `test_eth_client_lock_then_complete_transfer` to test counterpart contract call.

# Testing

I create a test that recreate the issue:

RUST_BACKTRACE=1 cargo test --test client_eth_movement test_eth_client_lock_then_complete_transfer -- --nocapture --test-threads=1 
# Outstanding issues

Still an issue with LockTransfer call that don't works

### CODE REVIEW

**Comment by andygolay** [comment]
So far:

- Eth -> Movement direction E2E test passes with `RUST_BACKTRACE=1 DOT_MOVEMENT_PATH="$(pwd)/.movement" cargo test --test bridge_e2e_test_framework test_bridge_transfer_eth_movement_happy_path -- --nocapture --test-threads=1`. Seems to be valid but could use another pair of eyes.

- I've added DO NOT MERGE to recent commits because:
  - in the Movement - Eth direction, I've modified `AtomicBridgeCounterpartyMOVE.sol` for debugging. Specifically:
    - `if (atomicBridgeInitiatorMOVE.poolBalance() < amount) revert InsufficientMOVEBalance();` is currently commented out in `lockBridgeTransfer.`
    - `atomicBridgeInitiatorMOVE.withdrawMOVE(details.recipient, details.amount);` is commented out in `completeBridgeTransfer`  
    - The above two lines MUST be uncommented and the Movement -> Eth direction test (`RUST_BACKTRACE=1 DOT_MOVEMENT_PATH="$(pwd)/.movement" cargo test --test bridge_e2e_test_framework test_bridge_transfer_movement_eth_happy_path -- --nocapture --test-threads=1`) MUST consistently pass before merging this PR.
  - I also modifying the Eth event monitoring for `counterpart_trlocked_event_filter` to have a timeout of 30 seconds for testing. It doesn't seem to help from what I can see.
- As of the latest commit, the relayer is flaky about receiving the Eth counterparty Locked event. Sometimes it does, sometimes it doesn't. I added logging in the test so in the testing terminal, the `LockDetails` are always outputted. So it seems that `lockBridgeTransfer` is successfully called on the Eth side, and the event is emitted. However the relayer isn't always picking it up.

File: protocol-units/bridge/contracts/src/AtomicBridgeCounterparty.sol
Comment (0xPrimata) [inline protocol-units/bridge/contracts/src/AtomicBridgeCounterparty.sol]: Please remove all console.log()s. They only serve for debugging and will weight on tx cost.

**Review by 0xPrimata — CHANGES_REQUESTED** [review]

**Comment by 0xmovses** [comment]
Some CI failing https://github.com/movementlabsxyz/movement/actions/runs/11911551471/job/33193286931?pr=690

**Review by 0xmovses — APPROVED** [review]

**Pull request merged**
